build-dl4j-stack.sh

```diff
@@ -70,26 +90,58 @@ if [ "$SCALAV" == "2.11" ]; then
   SCALA="2.11.7"
 fi
 
-cd ..
-
-# removes any existing repositories to ensure a clean build
-find . -maxdepth 1 -iname "libnd4j" -exec rm -rf "{}" \;
-find . -maxdepth 1 -iname "nd4j" -exec rm -rf "{}" \;
-find . -maxdepth 1 -iname "datavec" -exec rm -rf "{}" \;
-#find . -maxdepth 1 -iname "deeplearning4j" -exec rm -rf "{}" \;
+# set git cloning to a shallow depth if the option says so
+if [ -z $SHALLOW ]; then
+    GIT_CLONE="git clone"
+else
+    GIT_CLONE="git clone --depth 1"
+fi
 
 # Report argument values
-echo BUILD      = "${BUILD}"
-echo PACKAGING  = "${PACKAGING}"
-echo CHIP       = "${CHIP}"
-echo COMPUTE    = "${COMPUTE}"
-echo NATIVE     = "${NATIVE}"
-echo LIBTYPE    = "${LIBTYPE}"
-echo SCALAV     = "${SCALAV}"
+echo BUILD        = "${BUILD}"
+echo PACKAGING    = "${PACKAGING}"
+echo CHIP         = "${CHIP}"
+echo COMPUTE      = "${COMPUTE}"
+echo NATIVE       = "${NATIVE}"
+echo LIBTYPE      = "${LIBTYPE}"
+echo SCALAV       = "${SCALAV}"
+echo SHALLOW      = "${SHALLOW}"
+echo DELETE_REPOS = "${DELETE_REPOS}"
+echo TEST_ND4J    = "${TEST_ND4J}"
+echo TEST_DATAVEC = "${TEST_DATAVEC}"
+echo TEST_DL4J    = "${TEST_DL4J}"
+echo MVN_OPTS     = "${MVN_OPTS}"
+
+###########################
+# Script execution starts #
+###########################
+
+pushd ..
+
+# removes lingering snapshot artifacts from existing maven cache to ensure a
+# clean build
+JAVA_PROJECTS="nd4j datavec deeplearning4j"
+for dirName in $JAVA_PROJECTS; do
```

Comment:
Something that maybe out of scope here but worth acknowledging: Arbiter,rl4j, and other libs are needed as part of this as well.